internal/dataplane/translator/subtranslator/httproute.go

```diff
@@ -54,23 +54,52 @@ type HTTPRoutesTranslationResult struct {
 	HTTPRouteNameToTranslationErrors map[k8stypes.NamespacedName][]error
 }
 
+type splitHTTPRouteMatchesWithPrioritiesGroupedByRule map[string][]SplitHTTPRouteMatchToKongRoutePriority
+
 // TranslateHTTPRoutesToKongstateServices translates a set of HTTPRoutes to kongstate services,
 // and collect the translation errors in the process of translating.
 func TranslateHTTPRoutesToKongstateServices(
 	logger logr.Logger,
 	storer store.Storer,
 	routes []*gatewayapi.HTTPRoute,
 	combinedServicesFromDifferentHTTPRoutes bool,
+	expressionRoutes bool,
 ) HTTPRoutesTranslationResult {
 	serviceNameToRules := groupRulesFromHTTPRoutesByKongServiceName(routes, combinedServicesFromDifferentHTTPRoutes)
 
+	// When feature flag expression routes is enabled, we need first split the matches and assign priorities to them
+	// to set proper priorities to the translated Kong routes for satisfying the specification of priorities of HTTPRoute matches:
+	// https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1.HTTPRouteRule
+	ruleToSplitMatchesWithPriorities := make(splitHTTPRouteMatchesWithPrioritiesGroupedByRule)
+	if expressionRoutes {
+		ruleToSplitMatchesWithPriorities = groupHTTPRouteMatchesWithPrioritiesByRule(logger, routes)
+	}
```

Comment:
Can we move the allocation inside the `if` so that we don't pay the cost here when not using expression routes?

```suggestion
	var ruleToSplitMatchesWithPriorities splitHTTPRouteMatchesWithPrioritiesGroupedByRule
	if expressionRoutes {
		ruleToSplitMatchesWithPriorities = groupHTTPRouteMatchesWithPrioritiesByRule(logger, routes)
	}
```

Reply:
done